modyn/supervisor/internal/triggers/drift/decision_policy.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABC, abstractmethod
+from collections import deque
+
+from modyn.config.schema.pipeline.trigger.drift.metric import DynamicThresholdCriterion, ThresholdDecisionCriterion
+
+
+class DriftDecisionPolicy(ABC):
+    """Decision policy that will make the binary is_drift decisions based on
+    the similarity/distance metrics.
+
+    Each drift decision wraps one DriftMetric and observes its time
+    series of distance values.
+    """
+
+    @abstractmethod
+    def evaluate_decision(self, distance: float) -> bool:
+        """Evaluate the decision based on the distance value or the raw
+        is_drift decision.
+
+        Args:
+            distance: The distance value of the metric.
+
+        Returns:
+            The final is_drift decision.
+        """
+
+
+class ThresholdDecisionPolicy(DriftDecisionPolicy):
+    """Decision policy that will make the binary is_drift decisions based on a
+    threshold."""
+
+    def __init__(self, config: ThresholdDecisionCriterion):
+        self.config = config
+
+    def evaluate_decision(self, distance: float) -> bool:
+        return distance >= self.config.threshold
+
+
+class DynamicDecisionPolicy(DriftDecisionPolicy):
+    """Decision policy that will make the binary is_drift decisions based on a
+    dynamic threshold."""
```

Comment:
Maybe you could add one more sentence here explaining how the dynamic threshold is set, i.e., the logic around `num_more_extreme`